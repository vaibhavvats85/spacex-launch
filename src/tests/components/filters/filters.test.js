import { shallow } from 'enzyme';
import Filters from '../../../components/filters/filters';
import React from 'react';

let wrapper;
jest.mock('react-redux', () => ({
    useDispatch: () => jest.fn(),

}));
jest.mock('react-router-dom', () => ({
    useLocation: () => jest.fn(),
    useHistory: () => jest.fn()
}))
beforeEach(() => {
    wrapper = shallow(<Filters />);
});

test('should render filters component correctly', () => {
    expect(wrapper).toMatchSnapshot();
});
