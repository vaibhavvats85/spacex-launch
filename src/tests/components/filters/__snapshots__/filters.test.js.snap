// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render filters component correctly 1`] = `
<div
  className="filters"
>
  <h1>
    Filters
  </h1>
  <div
    className="type"
  >
    <span
      className="head"
    >
      Launch Year
    </span>
    <div
      className="btns"
    >
      <FilterButtons
        btns={
          Array [
            "2006",
            "2007",
            "2008",
            "2009",
            "2010",
            "2011",
            "2012",
            "2013",
            "2014",
            "2015",
            "2016",
            "2017",
            "2018",
            "2019",
            "2020",
          ]
        }
        clickEvent={[Function]}
        params={URLSearchParams {}}
        type="launch_year"
      />
    </div>
    <span
      className="head"
    >
      Successful Launch
    </span>
    <div
      className="btns"
    >
      <FilterButtons
        btns={
          Array [
            "True",
            "False",
          ]
        }
        clickEvent={[Function]}
        params={URLSearchParams {}}
        type="launch_success"
      />
    </div>
    <span
      className="head"
    >
      Successful Landing
    </span>
    <div
      className="btns"
    >
      <FilterButtons
        btns={
          Array [
            "True",
            "False",
          ]
        }
        clickEvent={[Function]}
        params={URLSearchParams {}}
        type="land_success"
      />
    </div>
  </div>
</div>
`;
